### PR TITLE
next: Role assign handler

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #668 

require #691

### CODE REVIEW

File: pkg/auth/handler/role_assign.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		return skyerr.NewInvalidArgument("unspecified roles in request", []string{"roles"})
+	}
+	if p.UserIDs == nil || len(p.UserIDs) == 0 {
+		return skyerr.NewInvalidArgument("unspecified users in request", []string{"users"})
+	}
+
+	return nil
+}
+
+// RoleAssignHandler allow system administractor to batch assign roles to
```
Comment (rickmak) [inline pkg/auth/handler/role_assign.go]: s/administractor/administrator/

File: pkg/auth/handler/role_assign.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// users
+//
+// RoleAssignHandler required user with admin role.
+// All specified users will assign to all roles specified. Roles not already
+// exisited in DB will be created. Users not already existed will be ignored.
```
Comment (rickmak) [inline pkg/auth/handler/role_assign.go]: s/exisited/existed/

File: pkg/auth/handler/role_assign.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+//
+// curl -X POST -H "Content-Type: application/json" \
+//   -d @- http://localhost:3000/ <<EOF
```
Comment (rickmak) [inline pkg/auth/handler/role_assign.go]: http://localhost:3000/role/assign

File: pkg/core/auth/authinfo/pq/store.go [hunk 4]
```diff
@@ -94,10 +98,9 @@ func (s AuthInfoStore) CreateAuth(authinfo *authinfo.AuthInfo) (err error) {
 		return skydb.ErrUserDuplicated
 	}
 
-	// TODO:
-	// if err := s.UpdateUserRoles(authinfo); err != nil {
-	// 	return skydb.ErrRoleUpdatesFailed
-	// }
+	if err := s.updateUserRoles(authinfo); err != nil {
+		return skydb.ErrRoleUpdatesFailed
+	}
```
Comment (carmenlau) [inline pkg/core/auth/authinfo/pq/store.go]: Please update the signup handler to assign default roles to authinfo when calling CreateAuth.

**Comment by Steven-Chan** [comment]
@carmenlau @rickmak updated

**Pull request merged**